src/main.rs

```diff
@@ -135,6 +135,20 @@ impl State {
         Ok(Self { key, crt, san })
     }
 
+    #[cfg(test)]
+    #[cfg(target_os = "wasi")]
+    pub fn load_bytes(key: &[u8], crt: &[u8]) -> anyhow::Result<Self> {
+        // Validate the syntax of the files.
+        PrivateKeyInfo::from_der(key)?;
+        Certificate::from_der(crt)?;
+
+        Ok(State {
+            crt: crt.to_vec(),
+            san: None,
+            key: key.to_vec().into(),
+        })
+    }
+
```

Comment:
I think what we want is `read(san: Option<String>, mut key: impl BufRead, mut crt: impl BufRead)`

That method would not need to be guarded by either `target_os` or `test`, instead - `State::load` should use that method directly after opening the files and constructing the buffered readers
This way we minimize the amount of duplication, amount of code we need to maintain *and* increase test coverage. (Note that parsing `pem` in the test would also be unnecessary)

Reply:
Done